contracts/core/SettingsManager.sol

```diff
@@ -490,32 +451,92 @@ contract SettingsManager is ISettingsManager, Ownable, Constants {
         return _master == _delegate || EnumerableSet.contains(_delegatesByMaster[_master], _delegate);
     }
 
-    function getFundingFee(
+    function getPnl(
         address _indexToken,
-        bool _isLong,
         uint256 _size,
-        uint256 _entryFundingRate
-    ) public view override returns (uint256) {
-        if (_size == 0) {
-            return 0;
+        uint256 _averagePrice,
+        uint256 _lastPrice,
+        int256 _fundingIndex,
+        bool _isLong
+    ) public view override returns (bool, uint256) {
+        require(_averagePrice > 0, "average price should be greater than zero");
+        int256 pnl;
+
+        if (_isLong) {
+            if (_lastPrice >= _averagePrice) {
+                pnl = int256((_size * (_lastPrice - _averagePrice)) / _averagePrice);
+            } else {
+                pnl = -1 * int256((_size * (_averagePrice - _lastPrice)) / _averagePrice);
+            }
+        } else {
+            if (_lastPrice <= _averagePrice) {
+                pnl = int256((_size * (_averagePrice - _lastPrice)) / _averagePrice);
+            } else {
+                pnl = -1 * int256((_size * (_lastPrice - _averagePrice)) / _averagePrice);
+            }
         }
 
-        uint256 fundingRate = cumulativeFundingRates[_indexToken][_isLong] - _entryFundingRate;
-        if (fundingRate == 0) {
-            return 0;
+        pnl += getFundingFee(_indexToken, _isLong, _size, _fundingIndex);
+
+        if (pnl > 0) {
+            return (true, uint256(pnl));
+        }
+        return (false, uint256(-1 * pnl));
+    }
+
+    function updateFunding(address _indexToken) external override {
+        if (lastFundingTimes[_indexToken] != 0) {
```

Comment:
why compare to zero? change to `if (lastFundingTimes[_indexToken] != block.timestamp)`?